Minutes 2021-07-21 (#240)
solid/authorization-panel: meetings/2021-07-21.md

Matthieu: But it is a security issue to have ACRs readable publicly by default.

Henry: It depends how it is done. 
It can be done right, and in my view, far from being an exceptional case, it is going to be the majority of cases that do this.

Justin: I think we're conflating a real problem that needs to be solved: which is understanding what a given agent has access to.  An access control system should just be telling us whether you have access to something or not. But there are other out-of-band ways to give agents a list of what they have access to.
There aren't patterns to tell an agent what they have access to.
I personally don't believe that the best way to deliver that information is through the ACL resources. I'm happy to be proven wrong.
I see it as a separate type of flow.
In other areas which aren't Solid specific but if your control rules are setup (OAuth/Firewalls...).
I'm wondering what is the minimum amount that people need to know.
We have precedent with WAC-ALLOW in Solid.

> But we need to be able to follow your nose through resources would be pretty problematic. Because you don't want to leak stuff.

Author: I think the idea behind that statement is:

> Using a follow your nose mechanism, that is requiring the dereferencing of access control resources to understand access requirement would be pretty problematic. Because you don't want to leak information.

> But we need to be able to follow your nose through resources would be pretty problematic. Because you don't want to leak stuff.

Reviewer: Tweaked it a bit; updated suggestion.

Matthieu: This sounds like a classic verifiable credential/key-based access where you just need to prove what set you pertain to.

### PR Workflow

Pavlik: There is a bit of a problem regarding who works on pull requests and who has to approve them. But it's pretty clear that ACP would be Matthieu, WAC Sarven, WAC+ Henry, but it would be good to have 1 person leading the proposal.

Matthieu: That sounds good, but what do you think would be necessary to make things clearer?

Henry: I think that's more or less what we've been doing.

Pavlik: We could have separate PRs for separate proposals.

Matthieu: I would prefer keeping a PR per use case. I think the current flow does work well to compare the implementation details.

Justin: +1 Pavlik's point or the spirit of it.
I feel like we needed the specs yesterday and I can't build what I want with the current state of WAC.
We need to optimize our workflow to get the proposal to a finished state; we need to get there fast.
The ecosystem needs us to get to a better more robust access control system sooner rather than later.

Henry: I agree that WAC got stuck where it was for 5 to 6 years.

> At least now, the simple Use Cases help us show the limitations of WAC and get feedback, e.g. from Sarven.

Reviewer: ```suggestion
At least now, the simple Use Cases help us show the limitations of WAC and get feedback, e.g., from Sarven.
```

Comparing it to ACP is helping us tweak things a little bit.
We write out detail by detail the things and figure out problems as we go along.

> We seem to have for example noticed a problem with ACP and WAC both having to copying data. This allows us to figuring out what the problems and disavantages are of each, how to solve them, and perhaps even end up with one system.

Reviewer: ```suggestion
We seem to have, for example, noticed a problem with ACP and WAC both having to copy data. This allows us to figure out the problems and disadvantages of each, how to solve them, and perhaps even end up with one system.
```

We have not been putting enough time in it, which is why it's very slow. 
We had the vacations slowing us down right now, and before that we spent 3 weeks discussing [UMA](https://github.com/solid/authorization-panel/blob/main/meetings/2021-06-09.md).
I suggest we commit this PR and continue with new PRs to continue building on it.
Matthieu reminded me right before Elf and Justin joined, that [two weeks ago](https://github.com/solid/authorization-panel/commit/e0b77c5976056c42ab7018ade0e766c1fec78d0b) we already agreed to merge and incrementally improve the UC3 PR.

> WAC is very easy to implement which is good at the begining. It has been advertised as a good decentralised access control solution, and it is if we tackle a few issues so that it meets expectations.

Reviewer: ```suggestion
WAC is very easy to implement which is good at the beginning. It has been advertised as a good decentralised access control solution, and it is if we tackle a few issues so that it meets expectations.
```

Justin: It is a way and beneficial.
But the best way is to have something in the field and get it used in the field.

> We need value from real; "in the field"; implementations and use.

Reviewer: ```suggestion
We need value from real, "in the field" implementations and use.
```

We need to see value from there.
That's an essential part of maturing specs and getting them to completion to have something out there that is close to completion.
I wanna ideally try to get there.

Henry: That's what I was doing 7 years ago in a startup I was with. I had built a server and clients . But the startup did not work out. Then it took a long time because something got stuck at MIT.

If you look at this UC3 PR, then you can see that WAC+ import solves a lot of issues.
And it might help solve things on the ACP side too.
It helps you avoid copying lots of data around.

> And answers many questions, such as "If you change the root ACR do changes get effected everywhere else?...

Reviewer: ```suggestion
And answers many questions, such as "If you change the root ACR, do changes take effect everywhere else?...
```

Justin: Absolutely.

Pavlik: Yes. Another part of my suggestion is not expecting someone else from commenting on your PRs to make progress.

Matthieu: Yes, you're perfectly right, that has been problematic in the past, I'll try to adjust to make faster progress and focus on important issues and questions during the panel.

Henry: Adjusting expectations of comments to move along faster sounds good. We'll keep it in check so that it doesn't get out of control.
Let's not lose the benefit of questions and feedback.

Matthieu: +1

### Closing outdated UCR issues

Justin: +1

Henry: I'll follow up on it on GitHub

### Transfer of issue 191

Justin: I might have permission to do it.

Matthieu: I'll ask Sarven otherwise.

## Actions

* Matthieu to submit UC0 on effective access control resource discovery.
* Matthieu to create issues:
    * How does an agent know all the resources it has access to?

>     * How are required credentials to present to access a resource advertised?

Reviewer: ```suggestion
    * How are the credential presentations required to access a resource advertised?
```

Author: @TallTed I don't know how the "to present got in there". What I meant is:

> How to discover which credentials are required to access a resource?

Does that formulation make sense to you?

Reviewer: @matthieubosquet -- What do you think of this --

> How can an agent discover what credentials are required to access a resource?

Author: Reads well to me! (is it "what credentials" or "which credentials"?)

Reviewer: Updated. ("which" is better.)

>     * Using extending access modes instead of to many apply policy predicates in ACP.

Reviewer: This doesn't seem clear to me. Might need some rewording.
```suggestion
    * Using extending access modes instead of too many apply policy predicates in ACP.
```

>     * Using extending access modes instead of too many apply policy predicates in ACP.

Author: ```suggestion
    * Using extended access modes (control read, control write...) instead of `apply` + `access` predicates in ACP.
```

* Matthieu to clarify whether WAC's ACLs are private by default.
* Matthieu to point Henry to the extended access control modes issue/proposal and Henry to review it.
* Henry and Matthieu to bootstrap the use case on privacy.
* Henry to follow up on GitHub on outdated issues #87 and #89

> * Matthieu to follow up on transfet of https://github.com/solid/authorization-panel/issues/191

Reviewer: ```suggestion
* Matthieu to follow up on transfer of https://github.com/solid/authorization-panel/issues/191
```

Matthieu: There are two problems here:
1. how does an agent know all the resources it has access to

> 2. how does an agent know which credentials to present to access a resource

Reviewer: ```suggestion
1. how does an agent know all the resources it has access to?
2. how does an agent know which credentials to present to access a resource?
```

Reviewer 2: I did not answer question 1. "how does an agent know all the resources it has access to?". Hmm...
